### ISSUE TITLE
Multiple NavMeshAgents of the same size share one NavMesh

### ISSUE THREAD

**Issue by Bunzaga** [Unity-Technologies/NavMeshComponents#15]
It would be nice if there was a way to have the navmesh surface allow more than a single agent type on its navmesh.

I am guessing it mostly has to do with the edges of the navmesh, and the radius of the agent?  For example, if an agent has a 0.5 unit radius, the navmesh generated is roughly 0.5 units away from everything.  If I then put on a 1.0 unit radius, he will overlap the edges of the mesh, and look like crap.

Couldn't the distance from edges be a separate value, and then the radius of the navmesh agent be part of the tunneling/distance from edge calculation?

For example, I could set the navmesh 'edge radius' to 0.25, meaning that when generated, it would be roughly 0.25 units away from all the geometry, etc.

Next, when calculating positioning, I would use something like...  agentRadius - navmeshEdgeRadius to determine how close to the edge of the navmesh I would allow.

In this way, if my edges are at 0.25, and my agents radius is 0.25, it would work like it normally does now.  If my agents radius is 0.125, it would overlap the edge a bit, but still look correct, as far as my agent goes.

If my agent is 1.0 radius I would really be 0.75 units away from the edge...

There may need to be some finagling for areas where the agents radius is larger than the navmesh path radius, for example, if the width of the path is only 0.5, but my agent has a radius of 1.0, the path should return 'blocked' to me, since my agent couldn't fit.

In any case, this is just an example, I'm sure the geniuses at Unity can come up with something that works.

TL;DR: Please come up with a way to have one navmesh surface, shared between more than one agent type, especially if those agent types share things like radius, height, etc.   Is this possible?

**Comment by jakobhu** [comment]
> Couldn't the distance from edges be a separate value, and then the radius of the navmesh agent be part of the tunneling/distance from edge calculation?

It could, and i tried this approach a couple of years ago -  but there are several drawbacks:

1) Robustness is difficult in highly tessellated regions.
2) Performance is worse - because the agent can no longer be considered a point.
3) No longer possible to see directly from the debug graphics if an agent is able to pass through a corridor or not. 

- so we gave up on that approach

**Comment by Bunzaga** [comment]
Ahh ok, glad to know it was at least considered :D I'm just having a hard time justifying creating three navmesh surfaces, one for 0.25 unit radius, one for 0.5 unit radius, and one for 1.0 unit radius.

It does seem to be very fast though, I guess it'll be a necessary evil, if I want to have a broad range of sizes.

Thank you!

**Comment by jakobhu** [comment]
Yes - it's the way to go for now. Be assured the runtime cost is lower this way (compared to dealing with over/under-sized agents on one navmesh).
